### PR TITLE
[GitHub] Raising 403 exception

### PR DESCRIPTION
**Pull request by parthpuri-elastic**
## Closes https://github.com/elastic/connectors-py/issues/2397

Raising 403 exception same as unauthorized exception in Github

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

## Release Note

Raises 403 as ForbiddenException in GitHub connector. Previously, the connector skipped the 403 status code.

### CODE REVIEW

**Comment by parthpuri-elastic** [comment]
buildkite test this

File: connectors/sources/github.py [hunk 1]
```diff
@@ -787,6 +791,9 @@ async def graphql(self, query, variables=None):
         except BadGraphQLRequest as exception:
             if self.get_rate_limit_encountered(exception.status_code, exception):
                 await self._put_to_sleep(resource_type="graphql")
+            elif exception.status == FORBIDDEN:
+                msg = "Provided GitHub token does not have the necessary permissions to perform the request."
```
Comment (seanstory) [inline connectors/sources/github.py:795]: Let's include the URL in the error message. Perhaps the query too, if that's not risky to have in logs?

**Review by seanstory — COMMENTED** [review]
Looks good. Minor nit

**Comment by parthpuri-elastic** [comment]
buildkite test this

**Review by seanstory — APPROVED** [review]

**Comment by seanstory** [comment]
Added backport versions. Please be careful @parthpuri-elastic when creating bugfix PRs to ensure they are backported all the way to the most recently released minor (currently 8.13)

**Comment by parthpuri-elastic** [comment]
buildkite test this

**Pull request merged**